### PR TITLE
CryptoTable and CoinPage

### PR DESCRIPTION
**Pull request by Basiol97**
-Multiple components have been refactored 
-The Linechart and Barchart components have been refactored so that they are now only presentation components.   -Integrated router v6 
-Added the CryptoTbale and CoinPage components.

### CODE REVIEW

File: src/components/CryptoTable/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        this.fetchMoreCoins();
+    }
+    fetchData = () => {
+        if (this.state.initialCoins < 50) {
+            this.setState({ initialCoins: this.state.initialCoins + 10 });
+        }
+    }
+
+    fetchMoreCoins = async () => {
+        const response = await fetch(`https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=${this.state.initialCoins}&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d`);
```
Comment (yoloOnTheBattlefield) [inline src/components/CryptoTable/index.js:29]: change the page not the per_page

File: src/components/Links/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import IconFeature from 'pages/CoinPage/assets/IconFeature.svg'
+import IconLink from 'pages/CoinPage/assets/IconLink.svg';
+import {
+    Link,
+    LinksContainer,
+    LinkContainer,
+    CopyIcon,
+} from 'pages/CoinPage/CoinPage.styled'
```
Comment (yoloOnTheBattlefield) [inline src/components/Links/index.js:9]: new line

install prettier

File: src/components/Links/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const Links = (props) => {
+    return (
+        <div>
+            <LinksContainer>
+                <LinkContainer>
+                    <div><Link target="_/" href={props.blockchainLink1}><img src={IconLink} alt={"IconLink"} /></Link></div>
+                    <div>{props.blockchainLink1}</div>
+                    <CopyIcon onClick={() => this.handleCopy(props.blockchainLink1)}><img src={IconFeature} alt={"IconFeature"} /></CopyIcon>
```
Comment (yoloOnTheBattlefield) [inline src/components/Links/index.js:17]: props.handleCopy

File: src/components/Links/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                </LinkContainer>
+                <LinkContainer>
+                    <div><Link target="_/" href={props.blockchainLink3}><img src={IconLink} alt={"IconLink"} /></Link></div>
+                    <div>{props.blockchainLink3}</div>
+                    <CopyIcon onClick={() => this.handleCopy(props.blockchainLink3)}><img src={IconFeature} alt={"IconFeature"} /></CopyIcon>
+                </LinkContainer>
+            </LinksContainer>
+        </div>
+    )
+}
```
Comment (yoloOnTheBattlefield) [inline src/components/Links/index.js:32]: create an array of links and map

File: src/components/Row/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    const colors = ["#FFB528", "#474C77", "#FE7D43", "#B3404A", "#74D0B4", "#2775C9"];
+    const colorIndex = Math.floor(Math.random(colors.length) * (colors.length - 0) + 0);
+    const backgroundColor = colors[colorIndex];
+
+    const formatSparkLineData = () => {
+        const sparkLineData = sparklineData?.price?.map((item, index) => {
+            if (index % 2 == 0) {
+                return item;
+            }
+        }).filter(function (element) {
```
Comment (yoloOnTheBattlefield) [inline src/components/Row/index.js:41]: return index % 2 == 0

Comment (yoloOnTheBattlefield) [reply src/components/Row/index.js:41]: data.filter(return index % 2 == 0 ).filter(Math.abs(index % 2) == 1)

File: src/components/SparkLine/SparkLine.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import { LineContainer } from 'pages/CoinList/CoinList.styled';
+
+const SparkLine = (props) => {
+
+    const labelArray = Array(42).fill(0);
+    const chartData = {
+        labels: labelArray,
+        datasets: [
+            {
+                borderColor: `${props.price_change_percentage_7d_in_currency > 0 ? "lightgreen" : "red"}`,
```
Comment (yoloOnTheBattlefield) [inline src/components/SparkLine/SparkLine.js:12]: props.price_change_percentage_7d_in_currency > 0 ? "lightgreen" : "red"

File: src/components/Summary/index.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+                </CoinPricesOuterContainer>
+                <CoinGeneralInfosContainer>
+                    <MarketCapContainer>
+                        <div><img src={IconlyBulkPlus} alt={"IconlyBulkPlus"} /></div><div>Market Cap:<span> ${coinFormatPrice(props.coinData.market_data?.market_cap?.usd)}</span><span>---%</span></div>
+                    </MarketCapContainer>
+                    <FullyDilutedContainer>
+                        <div><img src={IconlyBulkPlus} alt={"IconlyBulkPlus"} /></div><div>Fully Diluted Valuation:<span> ${coinFormatPrice(props.coinData.market_data?.fully_diluted_valuation?.usd)}</span></div>
+                    </FullyDilutedContainer>
+                    <VolumeContainer>
+                        <div><img src={IconlyBulkPlus} alt={"IconlyBulkPlus"} /></div><div>Volume 24h:<span> ${coinFormatPrice(props.coinData.market_data?.high_24h.usd)}</span></div>
```
Comment (yoloOnTheBattlefield) [inline src/components/Summary/index.js:110]: no need to use. {} if you have a string

File: src/components/Table/index.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    TableName,
+    TablePrice,
+    OneHour,
+    TwentyFourHours,
+    SevenDays,
+    VolMarketCap,
+    CirculatingTotalSup
+} from 'pages/CoinList/CoinList.styled'
+
+const Table = (props) => {
```
Comment (yoloOnTheBattlefield) [inline src/components/Table/index.js:17]: destructure here

File: src/components/Table/index.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                    image={item.image}
+                    current_price={item.current_price}
+                    total_volume={item.total_volume}
+                    market_cap={item.market_cap}
+                    total_supply={item.total_supply}
+                    circulating_supply={item.circulating_supply}
+                    price_change_percentage_1h_in_currency={item.price_change_percentage_1h_in_currency}
+                    price_change_percentage_24h_in_currency={item.price_change_percentage_24h_in_currency}
+                    price_change_percentage_7d_in_currency={item.price_change_percentage_7d_in_currency}
+                />)}
```
Comment (yoloOnTheBattlefield) [inline src/components/Table/index.js:52]: <Row index={index}
                 
                    key={item.id}  {...item}  />

File: src/pages/CoinPage/CoinPage.styled.js [hunk 10]
```diff
@@ -0,0 +1,381 @@
...
+}
+`;
+
+export const CoinContainer = styled.div`
+display: flex;
+flex-direction: column;
+justify-content: space-between;
+border-radius: 10px;
+width: 20%;
+
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage/CoinPage.styled.js:58]: indent and use prettier

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/components/Links/index.js (hunk 2 above)
Comment (Basiol97) [reply src/components/Links/index.js:9]: Resolved

File: src/components/Links/index.js (hunk 3 above)
Comment (Basiol97) [reply src/components/Links/index.js:17]: Resolved

File: src/components/Row/index.js (hunk 5 above)
Comment (Basiol97) [reply src/components/Row/index.js:41]: Resolved

File: src/components/SparkLine/SparkLine.js (hunk 6 above)
Comment (Basiol97) [reply src/components/SparkLine/SparkLine.js:12]: Resolved

File: src/components/Summary/index.js (hunk 7 above)
Comment (Basiol97) [reply src/components/Summary/index.js:110]: Resolved

File: src/components/Table/index.js (hunk 8 above)
Comment (Basiol97) [reply src/components/Table/index.js:17]: Resolved

File: src/components/Table/index.js (hunk 9 above)
Comment (Basiol97) [reply src/components/Table/index.js:52]: Resolved

File: src/components/CryptoTable/index.js (hunk 1 above)
Comment (Basiol97) [reply src/components/CryptoTable/index.js:29]: Resolved

File: src/components/Links/index.js (hunk 4 above)
Comment (Basiol97) [reply src/components/Links/index.js:32]: Resolved

File: src/pages/CoinPage/CoinPage.styled.js (hunk 10 above)
Comment (Basiol97) [reply src/pages/CoinPage/CoinPage.styled.js:58]: Resolved

**Pull request closed without merging**